test/mongodb.test.js

```diff
@@ -1406,6 +1406,37 @@ describe('mongodb connector', function() {
     });
   });
 
+  it('findOrCreate should properly support field projection (on create)', function() {
+    const query = {where: {title: 'Kopria post'}, fields: {comments: 1}};
+    const newData = {title: 'Kopria post', content: 'Xazo content', comments: ['comment1', 'comment2']};
+    return Post.findOrCreate(query, newData)
+      .then(function(result) {
+        const createdPost = result[0];
+        const created = result[1];
+        created.should.be.true();
+        should.not.exist(createdPost.title);
+        should.not.exist(createdPost.content);
+        createdPost.comments.should.containDeep(['comment1', 'comment2']);
+      });
+  });
+
+  it('findOrCreate should properly support field projection (on find)', function() {
```

Comment:
Can you add a similar test where `fields` is passed in as an array?

Reply:
done